### ISSUE TITLE
To do before `0.3.0`

### ISSUE THREAD

**Issue by IndrajeetPatil** [IndrajeetPatil/ggstatsplot#374]
- [x] Retire the `fill.gradient` option for `gghistostats`. 
It makes no sense to have fill gradient when the Y-axis labels are already showing the same info.

- [x] No need to have separate `broom::tidy` in `ggcoefstats` for mixed and non-mixed models

- [ ] Implement `outlier.point.args` for both `ggbetweenstats` and `ggwithinstats`

- [x] Implement `mean.path.args` and `point.path.args` for `ggwithinstats`

- [x] Separate statistical analysis return and plot return in `ggcorrmat`(same as the rest of the functions)

- [ ] Add lifecycle badges

- [ ] Think about `x` and `y` args in `ggbarstats` and `ggpiestats`
 
- [ ] For the love of god, have a more disciplined approach about the order in which different arguments are being catalogued for functions

- [x] Update `ggcoefstats` docs to reflect the fact that `LL` values are no longer included in the caption summary.

- [x] Skip palette usage in `ggscatterstats` - it just needs two colors

- [x] Change some `theme` defaults

- [ ] Retire `conf.int` argument for `ggcoefstats`

- [x] Rethink `ggdotplotstats` defaults

- [x] Retire sorting in `ggbetweenstats` and `ggwithinstats`; if users want to do this, they can do this outside

**Comment by IndrajeetPatil** [comment]
Skip palette usage in ggscatterstats 
It just needs two colors

**Comment by IndrajeetPatil** [comment]
Addressed in:  046a15d

**Comment by IndrajeetPatil** [comment]
How did this happen again?!

```r
ggstatsplot::ggcorrmat(
  data = ggplot2::msleep,
  cor.vars = sleep_total:bodywt,
  type = "spearman",
  output = "ci",
  p.adjust.method = "holm"
)
#> Note: In the correlation matrix,
#> the upper triangle: p-values adjusted for multiple comparisons
#> the lower triangle: unadjusted p-values.
```
